libraries/freertos_plus/standard/freertos_plus_tcp/source/FreeRTOS_DNS.c

```diff
@@ -378,126 +414,146 @@ typedef struct xDNSAnswerRecord DNSAnswerRecord_t;
 	/*-----------------------------------------------------------*/
 
 	/* A DNS reply was received, see if there is any matching entry and
-	call the handler. */
-	static void vDNSDoCallback( TickType_t xIdentifier, const char *pcName, uint32_t ulIPAddress );
-	static void vDNSDoCallback( TickType_t xIdentifier, const char *pcName, uint32_t ulIPAddress )
+	call the handler.  Returns pdTRUE if uxIdentifier was recognised. */
+	static BaseType_t xDNSDoCallback( TickType_t uxIdentifier,
+									  const char *pcName,
+									  uint32_t ulIPAddress );
```

Comment:
Is there any value to having a prototype immediately above it's function?

Reply:
It makes the code look less complicated.  The code for asynchronous callbacks is gathered in a big block :

~~~c
#if( ipconfigDNS_USE_CALLBACKS == 1 )
    /* 166 lines of code handling call-backs */
#endif
~~~

It is old code, and it is against the style rule that an `#ifdef` should include a single function.
In this block, a entire feature is included.
I don't mind to change this in a separate PR.

Hein